File: store/src/main/java/com/nytimes/android/external/store3/util/NoopParserFunc.java
package com.nytimes.android.external.store3.util;

import com.nytimes.android.external.store3.base.Parser;

import io.reactivex.annotations.NonNull;

/**
 * Pass-through parser for stores that parse externally
 */
public class NoopParserFunc<Raw, Parsed> implements Parser<Raw, Parsed> {

    @Override
    public Parsed apply(@NonNull Raw raw) throws ParserException {
        //noinspection unchecked
        return (Parsed) raw;
    }
}

File: store/src/main/java/com/nytimes/android/external/store3/util/NoopParserFunc.kt
package com.nytimes.android.external.store3.util

import com.nytimes.android.external.store3.base.Parser

import io.reactivex.annotations.NonNull

/**
 * Pass-through parser for stores that parse externally
 */
class NoopParserFunc<Raw, Parsed> : Parser<Raw, Parsed> {

    @Throws(ParserException::class)
    override
    suspend fun apply(raw: Raw): Parsed {

        return raw as Parsed
    }
}
